diff -ru libbfoverlay-9846185237f820c31b6b62e0bf586db2bb3c3966/configure.ac libbfoverlay-9846185237f820c31b6b62e0bf586db2bb3c3966.new/configure.ac
--- libbfoverlay-9846185237f820c31b6b62e0bf586db2bb3c3966/configure.ac	2023-10-24 05:15:26.000000000 +0000
+++ libbfoverlay-9846185237f820c31b6b62e0bf586db2bb3c3966.new/configure.ac	2023-10-27 08:58:26.040048242 +0000
@@ -1,4 +1,4 @@
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 AC_INIT(
  [libbfoverlay],
@@ -33,7 +33,7 @@
 
 dnl Support of internationalization (i18n)
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.21])
+AM_GNU_GETTEXT_VERSION([0.19])
 
 dnl Check for compiler language support
 AC_C_CONST
