--- nwipe-0.35/src/gui.c	2023-11-02 20:19:21.000000000 +0000
+++ nwipe-0.35.new/src/gui.c	2023-11-04 04:06:47.510600082 +0000
@@ -33,6 +33,8 @@
 #define _DEFAULT_SOURCE
 #endif
 
+#include <time.h>
+
 #ifndef _POSIX_SOURCE
 #define _POSIX_SOURCE
 #endif
@@ -42,7 +44,6 @@
 #include <ncurses.h>
 #include <panel.h>
 #include <stdint.h>
-#include <time.h>
 #include <libconfig.h>
 #include <string.h>
 #include <strings.h>
