diff -ru nwipe-0.35/configure.ac nwipe-0.35.new/configure.ac
--- nwipe-0.35/configure.ac	2023-11-02 21:19:21.000000000 +0100
+++ nwipe-0.35.new/configure.ac	2023-11-03 15:57:56.276506878 +0100
@@ -1,7 +1,7 @@
 #                                               -*- Autoconf -*-
 # Process this file with autoconf to produce a configure script.
 
-AC_PREREQ([2.71])
+AC_PREREQ([2.63])
 AC_INIT([nwipe],[0.35],[git@brumit.nl])
 AM_INIT_AUTOMAKE(foreign subdir-objects)
 AC_CONFIG_FILES([Makefile src/Makefile man/Makefile])
