diff -ru nwipe-0.37.old/src/prng.c nwipe-0.37.new/src/prng.c
--- nwipe-0.37.old/src/prng.c	2024-05-16 02:40:04.505840590 +0200
+++ nwipe-0.37.new/src/prng.c	2024-05-16 02:41:18.036207915 +0200
@@ -269,7 +269,7 @@
         *state = malloc( sizeof( add_lagg_fibonacci_state_t ) );
     }
     add_lagg_fibonacci_init(
-        (add_lagg_fibonacci_state_t*) *state, (unsigned long*) ( seed->s ), seed->length / sizeof( unsigned long ) );
+        (add_lagg_fibonacci_state_t*) *state, (uint64_t*) ( seed->s ), seed->length / sizeof( uint64_t ) );
 
     return 0;
 }
@@ -285,7 +285,7 @@
         *state = malloc( sizeof( xoroshiro256_state_t ) );
     }
     xoroshiro256_init(
-        (xoroshiro256_state_t*) *state, (unsigned long*) ( seed->s ), seed->length / sizeof( unsigned long ) );
+        (xoroshiro256_state_t*) *state, (uint64_t*) ( seed->s ), seed->length / sizeof( uint64_t ) );
 
     return 0;
 }
