diff -ru rizin-0.1.2.orig/meson.build rizin-0.1.2/meson.build
--- rizin-0.1.2.orig/meson.build	2021-03-31 00:21:50.223086649 +0200
+++ rizin-0.1.2/meson.build	2021-03-31 00:25:46.571404343 +0200
@@ -436,7 +436,6 @@
   'shlr/sdb/src/set.c',
   'shlr/sdb/src/cdb.c',
   'shlr/sdb/src/cdb_make.c',
-  'shlr/sdb/src/dict.c',
   'shlr/sdb/src/diff.c',
   'shlr/sdb/src/disk.c',
   'shlr/sdb/src/fmt.c',
@@ -474,7 +473,6 @@
   'shlr/sdb/src/set.h',
   'shlr/sdb/src/cdb_make.h',
   'shlr/sdb/src/config.h',
-  'shlr/sdb/src/dict.h',
   'shlr/sdb/src/ht_inc.h',
   'shlr/sdb/src/ht_pp.h',
   'shlr/sdb/src/ht_up.h',
