From 595721861283a7a2f7b22829b57e4f88fdcaed69 Mon Sep 17 00:00:00 2001
From: Michal Ambroz <723625+xambroz@users.noreply.github.com>
Date: Sat, 25 Nov 2023 21:05:58 +0100
Subject: [PATCH] next fails if string is not found in binary

fixes #2306
---
 pwnlib/commandline/libcdb.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/pwnlib/commandline/libcdb.py b/pwnlib/commandline/libcdb.py
index b32400d2c..767af5e5c 100644
--- a/pwnlib/commandline/libcdb.py
+++ b/pwnlib/commandline/libcdb.py
@@ -177,8 +177,11 @@ def translate_offset(offs, args, exe):
 
 def collect_synthetic_symbols(exe):
     available_symbols = ['str_bin_sh']
-    exe.symbols['str_bin_sh'] = next(exe.search(b'/bin/sh\x00'))
-
+    try:
+        exe.symbols['str_bin_sh'] = next(exe.search(b'/bin/sh\x00'))
+    except:
+        pass
+        
     libc_start_main_return = exe.libc_start_main_return
     if libc_start_main_return > 0:
         exe.symbols['__libc_start_main_ret'] = libc_start_main_return
