diff -ru libhmac-7ce99ac975e27be8e19eea9accf5ffce0304fe8a.old/configure.ac libhmac-7ce99ac975e27be8e19eea9accf5ffce0304fe8a.new/configure.ac
--- libhmac-7ce99ac975e27be8e19eea9accf5ffce0304fe8a.old/configure.ac	2024-04-17 20:55:59.000000000 +0200
+++ libhmac-7ce99ac975e27be8e19eea9accf5ffce0304fe8a.new/configure.ac	2024-05-18 17:28:15.795157910 +0200
@@ -149,16 +149,16 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
+dnl AC_CONFIG_FILES([libcerror/Makefile])
+dnl AC_CONFIG_FILES([libcthreads/Makefile])
 AC_CONFIG_FILES([libhmac/Makefile])
 AC_CONFIG_FILES([pyhmac/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
+dnl AC_CONFIG_FILES([libclocale/Makefile])
+dnl AC_CONFIG_FILES([libcnotify/Makefile])
+dnl AC_CONFIG_FILES([libcsplit/Makefile])
+dnl AC_CONFIG_FILES([libuna/Makefile])
+dnl AC_CONFIG_FILES([libcfile/Makefile])
+dnl AC_CONFIG_FILES([libcpath/Makefile])
 AC_CONFIG_FILES([hmactools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libhmac-7ce99ac975e27be8e19eea9accf5ffce0304fe8a.old/Makefile.am libhmac-7ce99ac975e27be8e19eea9accf5ffce0304fe8a.new/Makefile.am
--- libhmac-7ce99ac975e27be8e19eea9accf5ffce0304fe8a.old/Makefile.am	2024-04-17 20:55:59.000000000 +0200
+++ libhmac-7ce99ac975e27be8e19eea9accf5ffce0304fe8a.new/Makefile.am	2024-05-18 17:27:37.102963460 +0200
@@ -3,15 +3,7 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
 	libhmac \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
 	hmactools \
 	pyhmac \
 	po \
@@ -79,8 +79,8 @@
 
 library:
 	(cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) $(AM_MAKEFLAGS))
+#	(cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS))
+#	(cd $(srcdir)/libcthreads && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libhmac && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS))
 
