diff -ru libcsystem-e98764a93d94e330cf01a7e853d013d64ded5899.orig/configure.ac libcsystem-e98764a93d94e330cf01a7e853d013d64ded5899.new/configure.ac
--- libcsystem-e98764a93d94e330cf01a7e853d013d64ded5899.orig/configure.ac	2015-01-01 16:11:20.000000000 +0100
+++ libcsystem-e98764a93d94e330cf01a7e853d013d64ded5899.new/configure.ac	2015-06-08 05:04:27.509563599 +0200
@@ -143,11 +143,6 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
 AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libcsystem-e98764a93d94e330cf01a7e853d013d64ded5899.orig/Makefile.am libcsystem-e98764a93d94e330cf01a7e853d013d64ded5899.new/Makefile.am
--- libcsystem-e98764a93d94e330cf01a7e853d013d64ded5899.orig/Makefile.am	2015-01-01 16:11:20.000000000 +0100
+++ libcsystem-e98764a93d94e330cf01a7e853d013d64ded5899.new/Makefile.am	2015-06-08 05:04:10.833825926 +0200
@@ -3,11 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libclocale \
-	libcnotify \
-	libuna \
 	libcsystem \
 	po \
 	manuals \
@@ -58,11 +53,6 @@
 
 library:
 	cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcstring && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libclocale && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libuna && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/libcsystem && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS)
 
@@ -79,11 +69,6 @@
 	done && test -z "$$fail"
 
 splint:
-	(cd $(srcdir)/libcstring && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libclocale && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libcsystem && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
