diff -ru libfole-c6dbb26a2f2a531cd4d1101bbffa88cd110e3f90.000/configure.ac libfole-c6dbb26a2f2a531cd4d1101bbffa88cd110e3f90.001/configure.ac
--- libfole-c6dbb26a2f2a531cd4d1101bbffa88cd110e3f90.000/configure.ac	2025-01-20 13:24:47.884745053 +0100
+++ libfole-c6dbb26a2f2a531cd4d1101bbffa88cd110e3f90.001/configure.ac	2025-01-20 13:24:47.904745169 +0100
@@ -1,4 +1,4 @@
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 AC_INIT(
  [libfole],
@@ -33,7 +33,7 @@
 
 dnl Support of internationalization (i18n)
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.21])
+AM_GNU_GETTEXT_VERSION([0.19])
 
 dnl Check for compiler language support
 AC_C_CONST
