diff -ru oletools-0.60.2.orig/setup.py oletools-0.60.2.new/setup.py
--- oletools-0.60.2.orig/setup.py	2024-07-02 15:41:31.000000000 +0200
+++ oletools-0.60.2.new/setup.py	2024-07-10 01:16:47.709412477 +0200
@@ -103,6 +103,9 @@
     'oletools.thirdparty.xglob',
     'oletools.thirdparty.tablestream',
     'oletools.thirdparty.oledump',
+    'oletools.thirdparty.msoffcrypto',
+    'oletools.thirdparty.msoffcrypto.format',
+    'oletools.thirdparty.msoffcrypto.method',
 ]
 ##setupdir = '.'
 ##package_dir={'': setupdir}
@@ -190,6 +193,9 @@
     'oletools.thirdparty.DridexUrlDecoder': [
         'LICENSE.txt',
         ],
+    'oletools.thirdparty.msoffcrypto': [
+        'LICENSE.txt',
+        ],
     # 'oletools.thirdparty.tablestream': [
     #     'LICENSE', 'README',
     #     ],
@@ -329,7 +335,7 @@
             'colorclass',
             # msoffcrypto-tool is not installable on PyPy+Windows (see issue #473),
             # so we only require it if the platform is not Windows or not PyPy:
-            'msoffcrypto-tool; platform_python_implementation!="PyPy" or (python_version>="3" and platform_system!="Windows" and platform_system!="Darwin")',
+            #'msoffcrypto-tool; platform_python_implementation!="PyPy" or (python_version>="3" and platform_system!="Windows" and platform_system!="Darwin")',
             'pcodedmp>=1.2.5',
         ],
         extras_require={
