diff -ru oletools-b4b52d224d2fce4dafbcc1dca158cdd9ff2710f9/setup.py oletools-b4b52d224d2fce4dafbcc1dca158cdd9ff2710f9.new/setup.py
--- oletools-b4b52d224d2fce4dafbcc1dca158cdd9ff2710f9/setup.py	2017-06-15 22:15:26.000000000 +0200
+++ oletools-b4b52d224d2fce4dafbcc1dca158cdd9ff2710f9.new/setup.py	2017-06-16 18:46:26.905204844 +0200
@@ -41,7 +41,7 @@
 #--- METADATA -----------------------------------------------------------------
 
 name         = "oletools"
-version      = '0.51dev8'
+version      = '0.51dev11'
 desc         = "Python tools to analyze security characteristics of MS Office and OLE files (also called Structured Storage, Compound File Binary Format or Compound Document File Format), for Malware Analysis and Incident Response #DFIR"
 long_desc    = open('oletools/README.rst').read()
 author       = "Philippe Lagadec"
@@ -72,14 +72,9 @@
 packages=[
     "oletools",
     'oletools.thirdparty',
-    'oletools.thirdparty.olefile',
-    'oletools.thirdparty.easygui',
     'oletools.thirdparty.xxxswf',
-    'oletools.thirdparty.prettytable',
     'oletools.thirdparty.xglob',
     'oletools.thirdparty.DridexUrlDecoder',
-    'oletools.thirdparty.pyparsing',
-    'oletools.thirdparty.colorclass',
     'oletools.thirdparty.tablestream',
     'oletools.thirdparty.zipfile27',
 ]
@@ -157,31 +152,15 @@
         + rglob('oletools/doc', 'doc', '*.md')
         + rglob('oletools/doc', 'doc', '*.png'),
 
-    'oletools.thirdparty.olefile': [
-        'README.txt',
-        'LICENSE.txt',
-        ],
     'oletools.thirdparty.xglob': [
         'LICENSE.txt',
         ],
-    'oletools.thirdparty.easygui': [
-        'LICENSE.txt',
-        ],
     'oletools.thirdparty.xxxswf': [
         'LICENSE.txt',
         ],
-    'oletools.thirdparty.prettytable': [
-        'CHANGELOG', 'COPYING', 'README'
-        ],
     'oletools.thirdparty.DridexUrlDecoder': [
         'LICENSE.txt',
         ],
-    'oletools.thirdparty.pyparsing': [
-        'LICENSE', 'README',
-        ],
-    'oletools.thirdparty.colorclass': [
-        'LICENSE.txt',
-        ],
     'oletools.thirdparty.zipfile27': [
         'LICENSE.txt',
         ],
