diff -ru WhatWeb-0.5.1/Makefile WhatWeb-0.5.1.new/Makefile
--- WhatWeb-0.5.1/Makefile	2020-02-25 16:13:54.000000000 +0100
+++ WhatWeb-0.5.1.new/Makefile	2020-05-31 01:06:53.127055943 +0200
@@ -27,7 +27,8 @@
 
 	# copy whatweb into LIBPATH/NAME/ and create a symbolic link in the BINPATH
 	install -p $(INSTALLD) -m 755 $(NAME) $(DESTDIR)$(LIBPATH)/$(NAME)
-	ln -s $(DESTDIR)$(LIBPATH)/$(NAME)/$(NAME) $(DESTDIR)$(BINPATH)/$(NAME)
+	install -d $(DESTDIR)$(BINPATH)
+	ln -s $(LIBPATH)/$(NAME)/$(NAME) $(DESTDIR)$(BINPATH)/$(NAME)
 
 	cp -p -r my-plugins $(DESTDIR)$(LIBPATH)/$(NAME)/
 	cp -p -r plugins-disabled plugins lib plugin-development addons $(DESTDIR)$(LIBPATH)/$(NAME)/
