diff -ru libcdatetime-2a8fbb251eab3aca942532b31bbd2d1721b552ab/configure.ac libcdatetime-2a8fbb251eab3aca942532b31bbd2d1721b552ab.new/configure.ac
--- libcdatetime-2a8fbb251eab3aca942532b31bbd2d1721b552ab/configure.ac	2023-01-15 07:41:51.000000000 +0100
+++ libcdatetime-2a8fbb251eab3aca942532b31bbd2d1721b552ab.new/configure.ac	2023-06-28 00:57:14.963346618 +0200
@@ -90,7 +90,6 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libcdatetime/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libcdatetime-2a8fbb251eab3aca942532b31bbd2d1721b552ab/Makefile.am libcdatetime-2a8fbb251eab3aca942532b31bbd2d1721b552ab.new/Makefile.am
--- libcdatetime-2a8fbb251eab3aca942532b31bbd2d1721b552ab/Makefile.am	2023-01-15 07:41:51.000000000 +0100
+++ libcdatetime-2a8fbb251eab3aca942532b31bbd2d1721b552ab.new/Makefile.am	2023-06-28 00:54:01.442996360 +0200
@@ -3,7 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libcdatetime \
 	po \
 	manuals \
@@ -53,7 +52,6 @@
 
 library:
 	cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/libcdatetime && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS)
 
@@ -70,7 +68,6 @@
 	done && test -z "$$fail"
 
 splint:
-	(cd $(srcdir)/libcerror && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libcdatetime && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
