diff -ru beye-a0679f8263f09869e4e2826619d0310a04649ca0/configure.ac beye-a0679f8263f09869e4e2826619d0310a04649ca0.new/configure.ac
--- beye-a0679f8263f09869e4e2826619d0310a04649ca0/configure.ac	2021-10-07 15:04:28.000000000 +0200
+++ beye-a0679f8263f09869e4e2826619d0310a04649ca0.new/configure.ac	2023-11-04 23:30:27.088137610 +0100
@@ -5,7 +5,7 @@
 # Set VERSION so we only need to edit in one place (i.e., here)
 m4_define(BEYE_VERSION, [6.1.1.1.pre])
 
-AC_INIT([beye], BEYE_VERSION, [https://github.com/widgetii/beye])
+AC_INIT([beye],[BEYE_VERSION],[https://github.com/widgetii/beye])
 AC_CONFIG_AUX_DIR([.auto])
 AM_INIT_AUTOMAKE([foreign -Wall -Werror subdir-objects dist-bzip2])
 AC_CONFIG_HEADERS([bconfig.h])
@@ -30,6 +30,8 @@
   [  --enable-slang          slang graphics support (autodetected)])
 AC_ARG_ENABLE(ncurses,
   [  --enable-ncurses        ncurses graphics support (autodetected)])
+AC_ARG_ENABLE(gpm,
+  [  --enable-gpm            gpm mouse server (autodetected)])
 AC_ARG_ENABLE(iconv,
   [  --enable-iconv          iconv library support (autodetected)])
 
@@ -77,7 +79,7 @@
 
 # Broken?
 #AC_CHECK_HEADERS([endian.h],
-# AC_TRY_COMPILE([#include ], [switch (1) { case __LITTLE_ENDIAN: break; case __BYTE_ORDER: break; } ],[is_little_endian=0], [is_little_endian=1]),
+# AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include ]], [[switch (1) { case __LITTLE_ENDIAN: break; case __BYTE_ORDER: break; } ]])],[is_little_endian=0],[is_little_endian=1]),
 # AC_MSG_ERROR([Can't Determine Endianess])
 #)
 #if test $is_little_endian = 1 ; then
@@ -217,6 +219,22 @@
 #AC_CHECK_LIB(nsl, main)
 #fi
 
+if test "${enable_gpm}" != "no"; then
+  ac_cv_my_have_gpm="no"
+  AC_CHECK_HEADERS(gpm.h gpm/gpm.h,
+   [ac_cv_my_have_gpm="yes"
+    AC_DEFINE(_GPM_, 1, Define to 1 to activate the slang backend driver)
+    AC_CHECK_LIB(gpm, Gpm_Open,
+     [gpm="-lgpm"])
+    BEYE_LIBS="${BEYE_LIBS} -lgpm"
+    BEYE_DRIVERS="${BEYE_DRIVERS} gpm"
+    break])
+  if test "${ac_cv_my_have_gpm}" = "no" -a "${enable_gpm}" = "yes"; then
+    AC_MSG_ERROR([cannot find gpm development files])
+  fi
+fi
+
+
 if test "${enable_slang}" != "no"; then
   ac_cv_my_have_slang="no"
   AC_CHECK_HEADERS(slang.h slang/slang.h,
@@ -270,7 +288,7 @@
 
   AC_MSG_CHECKING([for iconv declaration])
   AC_CACHE_VAL(am_cv_proto_iconv, [
-      AC_TRY_COMPILE([
+      AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
   #include <stdlib.h>
   #include <iconv.h>
   extern
@@ -282,7 +300,7 @@
   #else
   size_t iconv();
   #endif
-  ], [], am_cv_proto_iconv_arg1="", am_cv_proto_iconv_arg1="const")
+  ]], [[]])],[am_cv_proto_iconv_arg1=""],[am_cv_proto_iconv_arg1="const"])
         am_cv_proto_iconv="extern size_t iconv (iconv_t cd, $am_cv_proto_iconv_arg1 char * *inbuf, size_t *inbytesleft, char * *outbuf, size_t *outbytesleft);"
         am_cv_func_iconv="yes"])
       am_cv_proto_iconv=`echo "[$]am_cv_proto_iconv" | tr -s ' ' | sed -e 's/( /(/'`
