diff -ru libfvalue-7357315fa55d5a30273aa78d98ca8e9c62d2cd8a.old/configure.ac libfvalue-7357315fa55d5a30273aa78d98ca8e9c62d2cd8a.new/configure.ac
--- libfvalue-7357315fa55d5a30273aa78d98ca8e9c62d2cd8a.old/configure.ac	2024-04-15 21:39:38.000000000 +0200
+++ libfvalue-7357315fa55d5a30273aa78d98ca8e9c62d2cd8a.new/configure.ac	2024-05-13 14:56:16.834654894 +0200
@@ -118,14 +118,6 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libfvalue-7357315fa55d5a30273aa78d98ca8e9c62d2cd8a.old/Makefile.am libfvalue-7357315fa55d5a30273aa78d98ca8e9c62d2cd8a.new/Makefile.am
--- libfvalue-7357315fa55d5a30273aa78d98ca8e9c62d2cd8a.old/Makefile.am	2024-04-15 21:39:38.000000000 +0200
+++ libfvalue-7357315fa55d5a30273aa78d98ca8e9c62d2cd8a.new/Makefile.am	2024-05-13 14:54:58.321355383 +0200
@@ -3,14 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libuna \
-	libfdatetime \
-	libfguid \
-	libfwnt \
 	libfvalue \
 	po \
 	manuals	\
@@ -67,14 +59,6 @@
 
 library:
 	(cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdatetime && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfguid && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfwnt && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libfvalue && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS))
 
