diff -ru libfvalue-a5b2497f680b4d656f80be82fc82abce1b2e0352.orig/configure.ac libfvalue-a5b2497f680b4d656f80be82fc82abce1b2e0352.new/configure.ac
--- libfvalue-a5b2497f680b4d656f80be82fc82abce1b2e0352.orig/configure.ac	2015-03-22 18:21:07.000000000 +0100
+++ libfvalue-a5b2497f680b4d656f80be82fc82abce1b2e0352.new/configure.ac	2015-06-08 04:51:58.463355286 +0200
@@ -148,15 +148,6 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libfvalue-a5b2497f680b4d656f80be82fc82abce1b2e0352.orig/Makefile.am libfvalue-a5b2497f680b4d656f80be82fc82abce1b2e0352.new/Makefile.am
--- libfvalue-a5b2497f680b4d656f80be82fc82abce1b2e0352.orig/Makefile.am	2015-03-22 18:21:07.000000000 +0100
+++ libfvalue-a5b2497f680b4d656f80be82fc82abce1b2e0352.new/Makefile.am	2015-06-08 04:51:40.343640675 +0200
@@ -3,15 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libuna \
-	libfdatetime \
-	libfguid \
-	libfwnt \
 	libfvalue \
 	po \
 	manuals	\
@@ -62,15 +53,6 @@
 
 library:
 	(cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcstring && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdatetime && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfguid && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfwnt && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libfvalue && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS))
 
@@ -87,15 +69,6 @@
 	done && test -z "$$fail"
 
 splint:
-	(cd $(srcdir)/libcstring && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdate && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdatetime && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfguid && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfwnt && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libfvalue && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
