diff -up aircrack-ng-1.0/src/airodump-ng.h.oui-path aircrack-ng-1.0/src/airodump-ng.h
--- aircrack-ng-1.0/src/airodump-ng.h.oui-path	2009-09-16 16:57:26.000000000 +0200
+++ aircrack-ng-1.0/src/airodump-ng.h	2009-09-16 16:57:53.000000000 +0200
@@ -70,7 +70,7 @@ static uchar ZERO[32] =
 "\x00\x00\x00\x00\x00\x00\x00\x00"
 "\x00\x00\x00\x00\x00\x00\x00\x00";
 
-#define OUI_PATH "/usr/local/etc/aircrack-ng/airodump-ng-oui.txt"
+#define OUI_PATH "/etc/aircrack-ng/airodump-ng-oui.txt"
 #define MIN_RAM_SIZE_LOAD_OUI_RAM 32768
 
 int read_pkts=0;
diff -up aircrack-ng-1.0/scripts/airodump-ng-oui-update.oui-path aircrack-ng-1.0/scripts/airodump-ng-oui-update
--- aircrack-ng-1.0/scripts/airodump-ng-oui-update.oui-path	2009-09-16 16:55:43.000000000 +0200
+++ aircrack-ng-1.0/scripts/airodump-ng-oui-update	2009-09-16 16:56:29.000000000 +0200
@@ -4,7 +4,7 @@ CURL=`which curl 2>/dev/null`
 WGET=`which wget 2>/dev/null`
 OUI_DOWNLOAD_URL="http://standards.ieee.org/regauth/oui/oui.txt"
 
-OUI_PATH="/usr/local/etc/aircrack-ng"
+OUI_PATH="/etc/aircrack-ng"
 AIRODUMP_NG_OUI="${OUI_PATH}/airodump-ng-oui.txt"
 OUI_IEEE="${OUI_PATH}/oui.txt"
 USERID=""
