diff -up binwalk-2.3.4/setup.py.oops binwalk-2.3.4/setup.py
--- binwalk-2.3.4/setup.py.oops	2023-02-01 11:15:53.000000000 -0500
+++ binwalk-2.3.4/setup.py	2023-02-02 20:28:44.655410043 -0500
@@ -12,7 +12,7 @@ except ImportError:
 from distutils.dir_util import remove_tree
 
 MODULE_NAME = "binwalk"
-MODULE_VERSION = "2.3.3"
+MODULE_VERSION = "2.3.4"
 SCRIPT_NAME = MODULE_NAME
 MODULE_DIRECTORY = os.path.dirname(os.path.realpath(__file__))
 
