diff -ru libcnotify-e4275c83baf9c2fe53a034b6a73bf34238428fc1/configure.ac libcnotify-e4275c83baf9c2fe53a034b6a73bf34238428fc1.new/configure.ac
--- libcnotify-e4275c83baf9c2fe53a034b6a73bf34238428fc1/configure.ac	2022-01-08 06:39:14.000000000 +0100
+++ libcnotify-e4275c83baf9c2fe53a034b6a73bf34238428fc1.new/configure.ac	2023-06-28 12:35:40.105573721 +0200
@@ -91,7 +91,6 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libcnotify/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libcnotify-e4275c83baf9c2fe53a034b6a73bf34238428fc1/Makefile.am libcnotify-e4275c83baf9c2fe53a034b6a73bf34238428fc1.new/Makefile.am
--- libcnotify-e4275c83baf9c2fe53a034b6a73bf34238428fc1/Makefile.am	2022-01-08 06:39:14.000000000 +0100
+++ libcnotify-e4275c83baf9c2fe53a034b6a73bf34238428fc1.new/Makefile.am	2023-06-28 12:35:17.433720818 +0200
@@ -3,7 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libcnotify \
 	po \
 	manuals \
@@ -53,7 +52,6 @@
 
 library:
 	cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS)
 
@@ -70,7 +68,6 @@
 	done && test -z "$$fail"
 
 splint:
-	(cd $(srcdir)/libcerror && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libcnotify && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
