diff -ru libcnotify-d1bd4e920356ecd29c66f4d21bb36fa2834dbf31.old/configure.ac libcnotify-d1bd4e920356ecd29c66f4d21bb36fa2834dbf31.new/configure.ac
--- libcnotify-d1bd4e920356ecd29c66f4d21bb36fa2834dbf31.old/configure.ac	2024-04-14 07:29:46.000000000 +0200
+++ libcnotify-d1bd4e920356ecd29c66f4d21bb36fa2834dbf31.new/configure.ac	2024-05-13 16:09:22.874620057 +0200
@@ -91,7 +91,6 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libcnotify/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libcnotify-d1bd4e920356ecd29c66f4d21bb36fa2834dbf31.old/Makefile.am libcnotify-d1bd4e920356ecd29c66f4d21bb36fa2834dbf31.new/Makefile.am
--- libcnotify-d1bd4e920356ecd29c66f4d21bb36fa2834dbf31.old/Makefile.am	2024-04-14 07:29:46.000000000 +0200
+++ libcnotify-d1bd4e920356ecd29c66f4d21bb36fa2834dbf31.new/Makefile.am	2024-05-13 16:09:22.875620061 +0200
@@ -3,7 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libcnotify \
 	po \
 	manuals \
@@ -60,7 +59,6 @@
 
 library:
 	cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS)
 
