diff -ru radare2-5.9.8.orig/shlr/qjs/src/quickjs.c radare2-5.9.8.new/shlr/qjs/src/quickjs.c
--- radare2-5.9.8.orig/shlr/qjs/src/quickjs.c	2024-11-19 12:38:30.000000000 +0100
+++ radare2-5.9.8.new/shlr/qjs/src/quickjs.c	2024-11-22 14:34:10.546707808 +0100
@@ -11311,6 +11311,8 @@
                    char dest[minimum_length(JS_ECVT_BUF_SIZE)],
                    size_t size, int *decpt)
 {
+    int i;
+
     if (n_digits == 0) {
         /* find the minimum number of digits (XXX: inefficient but simple) */
         // TODO(chqrlie) use direct method from quickjs-printf
@@ -11360,7 +11362,7 @@
                 return n_digits;    /* truncate the 2 extra digits */
         }
         /* round up in the string */
-        for(int i = n_digits;; i--) {
+        for(i = n_digits;; i--) {
             /* ignore the locale specific decimal point */
             if (is_digit(dest[i])) {
                 if (dest[i]++ < '9')
