diff -ru -N radare2-5.8.2/libr/syscall/d/gen.py radare2-5.8.2.new/libr/syscall/d/gen.py
--- radare2-5.8.2/libr/syscall/d/gen.py	1970-01-01 01:00:00.000000000 +0100
+++ radare2-5.8.2.new/libr/syscall/d/gen.py	2023-02-26 00:13:15.616002799 +0100
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+# args [sdbpath] [input] [output]
+import os
+import sys
+import subprocess
+
+
+try:
+	sdb_exe = sys.argv[1]
+	_input = sys.argv[2]
+	_output = sys.argv[3]
+	_tmpfile = _input + ".tmp"
+	with open(_input) as lines:
+		res = ""
+		for line in lines:
+			kv = line.rstrip().split("=", 1)
+			if kv[0] == "_":
+				res += line
+			else:
+				vv = kv[1].split(",")
+				res += vv[0] + "." + vv[1] + "=" + kv[0] + "\n"
+				res += line
+		with open(_tmpfile, "w") as file:
+			file.write(res)
+	subprocess.call([sdb_exe, _output, "==", _tmpfile])
+except Exception as e:
+	print(e)
+	print("Usage: gen.py [sdb_exe] [input] [output]")
+
diff -ru -N radare2-5.8.2/libr/syscall/d/meson.build radare2-5.8.2.new/libr/syscall/d/meson.build
--- radare2-5.8.2/libr/syscall/d/meson.build	2023-01-23 11:53:42.000000000 +0100
+++ radare2-5.8.2.new/libr/syscall/d/meson.build	2023-02-26 00:13:15.617002836 +0100
@@ -35,7 +35,7 @@
     gen_cmd = sdb_gen_cmd_cgen
   else
     outfile = '@0@.sdb'.format(file)
-    gen_cmd = sdb_gen_cmd
+    gen_cmd = syscall_sdb_gen_cmd
   endif
 
   r_syscall_d_sources += custom_target(outfile,
diff -ru -N radare2-5.8.2/meson.build radare2-5.8.2.new/meson.build
--- radare2-5.8.2/meson.build	2023-01-23 11:53:42.000000000 +0100
+++ radare2-5.8.2.new/meson.build	2023-02-26 00:13:15.619002909 +0100
@@ -1,6 +1,7 @@
 project('radare2', 'c', license : 'LGPL3', meson_version : '>=0.49', version : '5.8.2')
 
 py3_exe = import('python').find_installation('python3')
+gen_py = '@0@/libr/syscall/d/gen.py'.format(meson.current_source_dir())
 git_exe = find_program('git', required: false)
 pkgconfig_mod = import('pkgconfig')
 
@@ -553,6 +554,15 @@
   '@INPUT@'
 ]
 
+py_cmd = '__import__("os").readlink(r"@0@"))'.format(repo)
+syscall_sdb_gen_cmd = [
+  py3_exe,
+  gen_py,
+  sdb_exe,
+  '@INPUT@',
+  '@OUTPUT@',
+]
+
 # handle spp dependency
 spp_files = [
   'shlr/spp/spp.c'
