diff -ru libregf-9f9f9da3d3c9c36af784a07936dee6674bcd198c/configure.ac libregf-9f9f9da3d3c9c36af784a07936dee6674bcd198c.000/configure.ac
--- libregf-9f9f9da3d3c9c36af784a07936dee6674bcd198c/configure.ac	2024-08-19 06:40:21.000000000 +0200
+++ libregf-9f9f9da3d3c9c36af784a07936dee6674bcd198c.000/configure.ac	2025-01-20 01:51:25.480560007 +0100
@@ -173,20 +173,20 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
+dnl AC_CONFIG_FILES([libcerror/Makefile])
+dnl AC_CONFIG_FILES([libcthreads/Makefile])
+dnl AC_CONFIG_FILES([libcdata/Makefile])
+dnl AC_CONFIG_FILES([libclocale/Makefile])
+dnl AC_CONFIG_FILES([libcnotify/Makefile])
+dnl AC_CONFIG_FILES([libcsplit/Makefile])
+dnl AC_CONFIG_FILES([libuna/Makefile])
+dnl AC_CONFIG_FILES([libcfile/Makefile])
+dnl AC_CONFIG_FILES([libcpath/Makefile])
+dnl AC_CONFIG_FILES([libbfio/Makefile])
+dnl AC_CONFIG_FILES([libfcache/Makefile])
+dnl AC_CONFIG_FILES([libfdata/Makefile])
+dnl AC_CONFIG_FILES([libfdatetime/Makefile])
+dnl AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libregf/Makefile])
 AC_CONFIG_FILES([pyregf/Makefile])
 AC_CONFIG_FILES([regftools/Makefile])
diff -ru libregf-9f9f9da3d3c9c36af784a07936dee6674bcd198c/Makefile.am libregf-9f9f9da3d3c9c36af784a07936dee6674bcd198c.000/Makefile.am
--- libregf-9f9f9da3d3c9c36af784a07936dee6674bcd198c/Makefile.am	2024-08-19 06:40:21.000000000 +0200
+++ libregf-9f9f9da3d3c9c36af784a07936dee6674bcd198c.000/Makefile.am	2025-01-20 01:51:25.476559983 +0100
@@ -3,20 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfwnt \
 	libregf \
 	regftools \
 	pyregf \
@@ -85,19 +71,6 @@
 
 library:
 	(cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libclocale && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcsplit && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcfile && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcpath && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libbfio && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfcache && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdatetime && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfwnt && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libregf && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS))
 
