diff -ru libmapidb-4db7d4fed43b960f5fa60b47987cdc4d8e59df40.orig/configure.ac libmapidb-4db7d4fed43b960f5fa60b47987cdc4d8e59df40.new/configure.ac
--- libmapidb-4db7d4fed43b960f5fa60b47987cdc4d8e59df40.orig/configure.ac	2015-01-05 07:44:23.000000000 +0100
+++ libmapidb-4db7d4fed43b960f5fa60b47987cdc4d8e59df40.new/configure.ac	2015-07-01 02:48:19.051586966 +0200
@@ -123,9 +123,9 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
+dnl AC_CONFIG_FILES([libcstring/Makefile])
+dnl AC_CONFIG_FILES([libcerror/Makefile])
+dnl AC_CONFIG_FILES([libcnotify/Makefile])
 AC_CONFIG_FILES([libmapidb/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libmapidb-4db7d4fed43b960f5fa60b47987cdc4d8e59df40.orig/Makefile.am libmapidb-4db7d4fed43b960f5fa60b47987cdc4d8e59df40.new/Makefile.am
--- libmapidb-4db7d4fed43b960f5fa60b47987cdc4d8e59df40.orig/Makefile.am	2015-01-05 07:44:23.000000000 +0100
+++ libmapidb-4db7d4fed43b960f5fa60b47987cdc4d8e59df40.new/Makefile.am	2015-07-01 02:47:57.460916954 +0200
@@ -3,9 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcnotify \
 	libmapidb \
 	po \
 	manuals \
@@ -56,9 +53,6 @@
 
 library:
 	cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcstring && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/libmapidb && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS)
 
@@ -75,9 +69,6 @@
 	done && test -z "$$fail"
 
 splint:
-	(cd $(srcdir)/libcstring && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libmapidb && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
