diff --git a/libyara/Makefile.am b/libyara/Makefile.am
index 4fa363a..a9242b2 100644
--- a/libyara/Makefile.am
+++ b/libyara/Makefile.am
@@ -48,6 +48,7 @@ MODULES += modules/pe/pe_utils.c
 
 if CUCKOO_MODULE
 MODULES += modules/cuckoo/cuckoo.c
+MODULES += modules/androguard/androguard.c
 endif
 
 if MAGIC_MODULE
diff --git a/libyara/modules/module_list b/libyara/modules/module_list
index d81b8f3..ba43609 100644
--- a/libyara/modules/module_list
+++ b/libyara/modules/module_list
@@ -10,6 +10,7 @@ MODULE(dotnet)
 
 #ifdef CUCKOO_MODULE
 MODULE(cuckoo)
+MODULE(androguard)
 #endif
 
 #ifdef MAGIC_MODULE
