From d1a6ef20c049d86a136111dce53b4eb65c4df1bd Mon Sep 17 00:00:00 2001
From: "Victor M. Alvarez" <vmalvarez@virustotal.com>
Date: Fri, 20 Jan 2023 19:41:36 +0100
Subject: [PATCH] Fix another issue with `yr_le32toh` used incorrectly.

---
 libyara/modules/pe/pe.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libyara/modules/pe/pe.c b/libyara/modules/pe/pe.c
index 354c6a103..181a36879 100644
--- a/libyara/modules/pe/pe.c
+++ b/libyara/modules/pe/pe.c
@@ -912,8 +912,8 @@ static IMPORT_FUNCTION* pe_parse_import_descriptor(
         has_ordinal = 1;
       }
 
-      rva_address = yr_le32toh(
-          import_descriptor->FirstThunk + (sizeof(uint32_t) * func_idx));
+      rva_address = yr_le32toh(import_descriptor->FirstThunk) +
+                    (sizeof(uint32_t) * func_idx);
 
       if (name != NULL || has_ordinal == 1)
       {
