diff -ru libuna-0bec9356d6c9fd2684defa2807752e39c5d34014/configure.ac libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.new/configure.ac
--- libuna-0bec9356d6c9fd2684defa2807752e39c5d34014/configure.ac	2022-06-11 06:05:12.000000000 +0200
+++ libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.new/configure.ac	2023-06-28 12:51:21.830007907 +0200
@@ -119,12 +119,7 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcdatetime/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
 AC_CONFIG_FILES([unatools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libuna-0bec9356d6c9fd2684defa2807752e39c5d34014/Makefile.am libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.new/Makefile.am
--- libuna-0bec9356d6c9fd2684defa2807752e39c5d34014/Makefile.am	2022-06-11 06:05:12.000000000 +0200
+++ libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.new/Makefile.am	2023-06-28 12:51:00.356199872 +0200
@@ -3,12 +3,7 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libuna \
-	libcdatetime \
-	libclocale \
-	libcnotify \
-	libcfile \
 	unatools \
 	po \
 	documents \
@@ -60,7 +55,6 @@
 
 library:
 	cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/libuna && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS)
 
