diff -ru libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.new/configure.ac libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.bootstrap/configure.ac
--- libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.new/configure.ac	2023-06-28 12:51:21.830007907 +0200
+++ libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.bootstrap/configure.ac	2023-06-28 13:18:41.898461872 +0200
@@ -120,7 +120,6 @@
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
 AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([unatools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
 AC_CONFIG_FILES([documents/Makefile])
diff -ru libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.new/Makefile.am libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.bootstrap/Makefile.am
--- libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.new/Makefile.am	2023-06-28 12:51:00.356199872 +0200
+++ libuna-0bec9356d6c9fd2684defa2807752e39c5d34014.bootstrap/Makefile.am	2023-06-28 13:18:15.357465089 +0200
@@ -4,7 +4,6 @@
 	include \
 	common \
 	libuna \
-	unatools \
 	po \
 	documents \
 	manuals \
@@ -72,7 +71,6 @@
 
 splint:
 	(cd $(srcdir)/libuna && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/unatools && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/tests && $(MAKE) splint $(AM_MAKEFLAGS))
 
