diff -u --recursive scapy-2.4.3-vanilla/scapy/arch/bpf/core.py scapy-2.4.3/scapy/arch/bpf/core.py
--- scapy-2.4.3-vanilla/scapy/arch/bpf/core.py	2019-08-04 11:12:30.000000000 -0500
+++ scapy-2.4.3/scapy/arch/bpf/core.py	2021-03-11 13:30:35.033061720 -0600
@@ -28,7 +28,7 @@
 
 # ctypes definitions
 
-LIBC = cdll.LoadLibrary(find_library("libc"))
+LIBC = cdll.LoadLibrary(find_library("c"))
 LIBC.ioctl.argtypes = [c_int, c_ulong, c_char_p]
 LIBC.ioctl.restype = c_int
 
