diff -ru libfdata-1d2e7f0edff9d676ccc3e96d40926d83d63af1e5.orig/configure.ac libfdata-1d2e7f0edff9d676ccc3e96d40926d83d63af1e5.new/configure.ac
--- libfdata-1d2e7f0edff9d676ccc3e96d40926d83d63af1e5.orig/configure.ac	2015-04-12 16:28:39.000000000 +0200
+++ libfdata-1d2e7f0edff9d676ccc3e96d40926d83d63af1e5.new/configure.ac	2015-06-08 01:49:04.330893351 +0200
@@ -139,11 +139,6 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
 AC_CONFIG_FILES([libfdata/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libfdata-1d2e7f0edff9d676ccc3e96d40926d83d63af1e5.orig/Makefile.am libfdata-1d2e7f0edff9d676ccc3e96d40926d83d63af1e5.new/Makefile.am
--- libfdata-1d2e7f0edff9d676ccc3e96d40926d83d63af1e5.orig/Makefile.am	2015-04-12 16:28:39.000000000 +0200
+++ libfdata-1d2e7f0edff9d676ccc3e96d40926d83d63af1e5.new/Makefile.am	2015-06-08 01:48:12.062726493 +0200
@@ -3,11 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libfcache \
 	libfdata \
 	po \
 	manuals \
@@ -59,11 +53,6 @@
 
 library:
 	(cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfcache && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libfdata && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS))
 
