diff -ru libclocale-45b989bec6c5578b762f5b13dbce03aa2e999426.old/configure.ac libclocale-45b989bec6c5578b762f5b13dbce03aa2e999426.new/configure.ac
--- libclocale-45b989bec6c5578b762f5b13dbce03aa2e999426.old/configure.ac	2024-04-14 06:27:18.000000000 +0200
+++ libclocale-45b989bec6c5578b762f5b13dbce03aa2e999426.new/configure.ac	2024-05-13 16:16:09.437468453 +0200
@@ -94,7 +94,6 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
 AC_CONFIG_FILES([libclocale/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libclocale-45b989bec6c5578b762f5b13dbce03aa2e999426.old/Makefile.am libclocale-45b989bec6c5578b762f5b13dbce03aa2e999426.new/Makefile.am
--- libclocale-45b989bec6c5578b762f5b13dbce03aa2e999426.old/Makefile.am	2024-04-14 06:27:18.000000000 +0200
+++ libclocale-45b989bec6c5578b762f5b13dbce03aa2e999426.new/Makefile.am	2024-05-13 16:16:09.439468463 +0200
@@ -3,7 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
 	libclocale \
 	po \
 	manuals \
@@ -60,7 +59,6 @@
 
 library:
 	cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/libclocale && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS)
 
