diff -ru libsmraw-995b9e200ae8c836439f14741253143803fecb0c.old/configure.ac libsmraw-995b9e200ae8c836439f14741253143803fecb0c.new/configure.ac
--- libsmraw-995b9e200ae8c836439f14741253143803fecb0c.old/configure.ac	2024-06-03 13:58:01.861751218 +0200
+++ libsmraw-995b9e200ae8c836439f14741253143803fecb0c.new/configure.ac	2024-06-03 14:02:14.821039215 +0200
@@ -173,23 +173,9 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
 AC_CONFIG_FILES([libsmraw/Makefile])
 AC_CONFIG_FILES([pysmraw/Makefile])
 AC_CONFIG_FILES([smraw.net/Makefile])
-AC_CONFIG_FILES([libhmac/Makefile])
 AC_CONFIG_FILES([smrawtools/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libsmraw-995b9e200ae8c836439f14741253143803fecb0c.old/Makefile.am libsmraw-995b9e200ae8c836439f14741253143803fecb0c.new/Makefile.am
--- libsmraw-995b9e200ae8c836439f14741253143803fecb0c.old/Makefile.am	2024-06-03 13:57:19.088533416 +0200
+++ libsmraw-995b9e200ae8c836439f14741253143803fecb0c.new/Makefile.am	2024-06-03 14:01:38.901856331 +0200
@@ -3,21 +3,7 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfvalue \
 	libsmraw \
-	libhmac \
 	smrawtools \
 	smraw.net \
 	pysmraw \
@@ -86,19 +72,6 @@
 
 library:
 	(cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libclocale && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcsplit && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcfile && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcpath && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libbfio && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfcache && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfvalue && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libsmraw && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS))
 
