diff -ru dionaea-0.7.0/modules/python/module.c dionaea-0.7.0.new/modules/python/module.c
--- dionaea-0.7.0/modules/python/module.c	2024-06-14 11:45:44.206540637 +0200
+++ dionaea-0.7.0.new/modules/python/module.c	2024-06-14 11:47:32.103089823 +0200
@@ -157,7 +157,7 @@
 		Py_DECREF(module);
 		PyObject *func = PyObject_GetAttrString(module, "remoteshell");
 		PyObject *arglist = Py_BuildValue("()");
-		PyObject *r = PyEval_CallObject(func, arglist);
+		PyObject *r = PyObject_CallObject(func, arglist);
 		Py_DECREF(arglist);
 		g_debug("r %p", r);
 		struct head 
@@ -200,7 +200,7 @@
 			PyObject *func = PyObject_GetAttrString(i->module, "stop");
 			if ( func != NULL ) {
 				PyObject *arglist = Py_BuildValue("()");
-				PyObject *r = PyEval_CallObject(func, arglist);
+				PyObject *r = PyObject_CallObject(func, arglist);
 				traceback();
 				// PyErr_Print();
 				Py_DECREF(arglist);
@@ -222,7 +222,7 @@
 			func = PyObject_GetAttrString(module, "new");
 			if( func != NULL ) {
 				PyObject *arglist = Py_BuildValue("()");
-				PyObject *r = PyEval_CallObject(func, arglist);
+				PyObject *r = PyObject_CallObject(func, arglist);
 				traceback();
 				Py_DECREF(arglist);
 				Py_XDECREF(r);
@@ -233,7 +233,7 @@
 			func = PyObject_GetAttrString(module, "start");
 			if( func != NULL ) {
 				PyObject *arglist = Py_BuildValue("()");
-				PyObject *r = PyEval_CallObject(func, arglist);
+				PyObject *r = PyObject_CallObject(func, arglist);
 				traceback();
 				Py_DECREF(arglist);
 				Py_XDECREF(r);
@@ -258,7 +258,7 @@
 			PyObject *func = PyObject_GetAttrString(module, "new");
 			if( func != NULL ) {
 				PyObject *arglist = Py_BuildValue("()");
-				PyObject *r = PyEval_CallObject(func, arglist);
+				PyObject *r = PyObject_CallObject(func, arglist);
 				Py_DECREF(arglist);
 				Py_XDECREF(r);
 				Py_DECREF(func);
@@ -292,7 +292,7 @@
 		if( func != NULL )
 		{
 			PyObject *arglist = Py_BuildValue("()");
-			PyObject *r = PyEval_CallObject(func, arglist);
+			PyObject *r = PyObject_CallObject(func, arglist);
 			Py_DECREF(arglist);
 			Py_XDECREF(r);
 			Py_DECREF(func);
@@ -323,7 +323,7 @@
 		if( func != NULL )
 		{
 			PyObject *arglist = Py_BuildValue("()");
-			PyObject *r = PyEval_CallObject(func, arglist);
+			PyObject *r = PyObject_CallObject(func, arglist);
 			Py_DECREF(arglist);
 			Py_XDECREF(r);
 			Py_DECREF(func);
@@ -401,7 +401,7 @@
 		PyObject *func = PyObject_GetAttrString(module, "new");
 		if( func != NULL ) {
 			PyObject *arglist = Py_BuildValue("()");
-			PyObject *r = PyEval_CallObject(func, arglist);
+			PyObject *r = PyObject_CallObject(func, arglist);
 			Py_DECREF(arglist);
 			Py_XDECREF(r);
 			Py_DECREF(func);
