Instead of using embedded pyev, use the system one.
diff -ru dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/dionaea/hpfeeds.py dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/dionaea/hpfeeds.py
--- dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/dionaea/hpfeeds.py	2018-03-21 04:21:30.195354666 +0100
+++ dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/dionaea/hpfeeds.py	2018-03-21 06:59:11.422191102 +0100
@@ -38,7 +38,7 @@
 from time import gmtime, strftime
 
 try:
-    from dionaea import pyev
+    import pyev
 except ImportError:
     pyev = None
 
diff -ru dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/dionaea/sip/__init__.py dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/dionaea/sip/__init__.py
--- dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/dionaea/sip/__init__.py	2018-03-21 04:21:30.198354625 +0100
+++ dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/dionaea/sip/__init__.py	2018-03-21 07:00:05.553401019 +0100
@@ -36,9 +36,10 @@
 import os
 import datetime
 import tempfile
+import pyev
 
 from dionaea.core import connection, g_dionaea, incident
-from dionaea import pyev, ServiceLoader
+from dionaea import ServiceLoader
 
 from dionaea.sip.extras import msg_to_icd, SipConfig, ErrorWithResponse
 
diff -ru dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/dionaea/submit_http.py dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/dionaea/submit_http.py
--- dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/dionaea/submit_http.py	2018-03-21 04:21:30.196354652 +0100
+++ dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/dionaea/submit_http.py	2018-03-21 06:58:39.522656697 +0100
@@ -1,7 +1,8 @@
 from dionaea.core import ihandler, incident, g_dionaea
 from dionaea.util import md5file, sha512file
-from dionaea import pyev, IHandlerLoader
+from dionaea import IHandlerLoader
 
+import pyev
 import logging
 import uuid
 import struct
diff -ru dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/dionaea/virustotal.py dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/dionaea/virustotal.py
--- dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/dionaea/virustotal.py	2018-03-21 04:21:30.196354652 +0100
+++ dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/dionaea/virustotal.py	2018-03-21 06:59:41.793747808 +0100
@@ -32,7 +32,7 @@
 import json
 import uuid
 import sqlite3
-from dionaea import pyev
+import pyev
 
 logger = logging.getLogger('virustotal')
 logger.setLevel(logging.DEBUG)
diff -ru dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/Makefile.am dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/Makefile.am
--- dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/Makefile.am	2018-03-21 04:21:30.195354666 +0100
+++ dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/Makefile.am	2018-03-21 07:14:31.670155062 +0100
@@ -10,13 +10,11 @@
 	$(PYTHON) setup.py build
 
 install-exec-am: all
-	cp build/*/dionaea/*.so $(DESTDIR)$(pkglibdir)/python.so
-	(cd $(DESTDIR)$(pkglibdir)/python/dionaea && rm -f core.so && ln -s ../../python.so core.so)
-	(cd $(DESTDIR)$(pkglibdir)/python/dionaea && rm -f pyev.so && ln -s ../../python.so pyev.so)
+	$(PYTHON) setup.py install -O1 --skip-build --root $(DESTDIR)/
 
 clean:
 	$(PYTHON) setup.py clean
-	rm -rf ./build binding.{c,h} 
+	rm -rf ./build binding.{c,h}
 
 dist-clean: clean
 
Only in dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python: pyev
diff -ru dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/setup.py.in dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/setup.py.in
--- dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.orig/modules/python/setup.py.in	2018-03-21 10:40:22.942127530 +0100
+++ dionaea-793accd84432a77309fa8b81e1f5e9b5bd9ee7a3.new/modules/python/setup.py.in	2018-03-21 10:41:35.418132420 +0100
@@ -81,7 +81,7 @@
 
 ext_modules=[
 	Extension("dionaea.core", 
-		['binding.pyx', 'module.c', 'pyev/pyev.c'], 
+		['binding.pyx', 'module.c'],
 		language="c",
 		include_dirs=['../../include', '../../'] + [k for k in sorted(include_dir_dict)],
 		extra_compile_args=[k for k in sorted(extra_compile_dict)],
