diff --git a/modules/python/Makefile.am b/modules/python/Makefile.am
index 5beb169..aa273d3 100644
--- a/modules/python/Makefile.am
+++ b/modules/python/Makefile.am
@@ -4,19 +4,20 @@ AUTOMAKE_OPTIONS = foreign
 
 SUBDIRS = dionaea util
 
+PYTHON_SITEARCH=$(pkglibdir)"/python"
+
 all: binding.c
 
 binding.c: setup.py setup.py.in binding.pyx module.c module.h ../../config.h ../../include/protocol.h ../../include/dionaea.h
 	$(PYTHON) setup.py build
 
 install-exec-am: all
-	cp build/*/dionaea/*.so $(DESTDIR)$(pkglibdir)/python.so
-	(cd $(DESTDIR)$(pkglibdir)/python/dionaea && rm -f core.so && ln -s ../../python.so core.so)
-	(cd $(DESTDIR)$(pkglibdir)/python/dionaea && rm -f pyev.so && ln -s ../../python.so pyev.so)
+	$(PYTHON) setup.py install -O1 --skip-build --root $(DESTDIR) --install-lib $(PYTHON_SITEARCH)
+	(cd $(DESTDIR)$(PYTHON_SITEARCH)/dionaea && export X=core*.so && ln -s $(PYTHON_SITEARCH)/dionaea/$$X python.so && mv python.so $(DESTDIR)$(pkglibdir)/python.so)
 
 clean:
 	$(PYTHON) setup.py clean
-	rm -rf ./build binding.{c,h} 
+	rm -rf ./build binding.{c,h}
 
 dist-clean: clean
 
diff --git a/modules/python/dionaea/hpfeeds.py b/modules/python/dionaea/hpfeeds.py
index 3674f4b..9a92e0d 100644
--- a/modules/python/dionaea/hpfeeds.py
+++ b/modules/python/dionaea/hpfeeds.py
@@ -38,7 +38,7 @@ import datetime
 from time import gmtime, strftime
 
 try:
-    from dionaea import pyev
+    import pyev
 except ImportError:
     pyev = None
 
diff --git a/modules/python/dionaea/sip/__init__.py b/modules/python/dionaea/sip/__init__.py
index 66cb86e..48100f0 100644
--- a/modules/python/dionaea/sip/__init__.py
+++ b/modules/python/dionaea/sip/__init__.py
@@ -36,9 +36,10 @@ import random
 import os
 import datetime
 import tempfile
+import pyev
 
 from dionaea.core import connection, g_dionaea, incident
-from dionaea import pyev, ServiceLoader
+from dionaea import ServiceLoader
 
 from dionaea.sip.extras import msg_to_icd, SipConfig, ErrorWithResponse
 
diff --git a/modules/python/dionaea/submit_http.py b/modules/python/dionaea/submit_http.py
index 20d645d..66c166c 100644
--- a/modules/python/dionaea/submit_http.py
+++ b/modules/python/dionaea/submit_http.py
@@ -1,7 +1,8 @@
 from dionaea.core import ihandler, incident, g_dionaea
 from dionaea.util import md5file, sha512file
-from dionaea import pyev, IHandlerLoader
+from dionaea import IHandlerLoader
 
+import pyev
 import logging
 import uuid
 import struct
diff --git a/modules/python/dionaea/virustotal.py b/modules/python/dionaea/virustotal.py
index f2508a7..3e96c3d 100644
--- a/modules/python/dionaea/virustotal.py
+++ b/modules/python/dionaea/virustotal.py
@@ -32,7 +32,7 @@ import logging
 import json
 import uuid
 import sqlite3
-from dionaea import pyev
+import pyev
 
 logger = logging.getLogger('virustotal')
 logger.setLevel(logging.DEBUG)
diff --git a/modules/python/setup.py.in b/modules/python/setup.py.in
index 17d71c0..97da738 100644
--- a/modules/python/setup.py.in
+++ b/modules/python/setup.py.in
@@ -86,7 +86,7 @@ PYEV_VERSION = "\"{0}\"".format(pyev_version)
 
 ext_modules=[
 	Extension("dionaea.core", 
-		['binding.pyx', 'module.c', 'pyev/pyev.c'], 
+		['binding.pyx', 'module.c'],
 		language="c",
 		include_dirs=['../../include', '../../'] + [k for k in sorted(include_dir_dict)],
 		extra_compile_args=[k for k in sorted(extra_compile_dict)],
