diff -ru libesedb-a349807aa4dbdd0bf46bed24af9c8a93ce02f9a5.orig/configure.ac libesedb-a349807aa4dbdd0bf46bed24af9c8a93ce02f9a5.new/configure.ac
--- libesedb-a349807aa4dbdd0bf46bed24af9c8a93ce02f9a5.orig/configure.ac	2015-04-12 19:36:54.000000000 +0200
+++ libesedb-a349807aa4dbdd0bf46bed24af9c8a93ce02f9a5.new/configure.ac	2015-06-05 02:13:19.731962924 +0200
@@ -202,27 +202,7 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libesedb/Makefile])
-AC_CONFIG_FILES([libfmapi/Makefile])
-AC_CONFIG_FILES([libmapidb/Makefile])
-AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([esedbtools/Makefile])
 AC_CONFIG_FILES([pyesedb/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
diff -ru libesedb-a349807aa4dbdd0bf46bed24af9c8a93ce02f9a5.orig/Makefile.am libesedb-a349807aa4dbdd0bf46bed24af9c8a93ce02f9a5.new/Makefile.am
--- libesedb-a349807aa4dbdd0bf46bed24af9c8a93ce02f9a5.orig/Makefile.am	2015-04-12 19:36:54.000000000 +0200
+++ libesedb-a349807aa4dbdd0bf46bed24af9c8a93ce02f9a5.new/Makefile.am	2015-06-05 02:12:12.802058562 +0200
@@ -3,27 +3,7 @@
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfguid \
-	libfvalue \
-	libfwnt \
 	libesedb \
-	libfmapi \
-	libmapidb \
-	libcsystem \
 	esedbtools \
 	pyesedb \
 	po \
