diff -ru libewf-5e47a75e30a4dd3068f8210f0a62c37b58329d7f.old/configure.ac libewf-5e47a75e30a4dd3068f8210f0a62c37b58329d7f.new/configure.ac
--- libewf-5e47a75e30a4dd3068f8210f0a62c37b58329d7f.old/configure.ac	2015-01-26 08:31:34.000000000 +0100
+++ libewf-5e47a75e30a4dd3068f8210f0a62c37b58329d7f.new/configure.ac	2015-06-30 19:42:20.034086204 +0200
@@ -218,28 +218,28 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcdatetime/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
-AC_CONFIG_FILES([libhmac/Makefile])
-AC_CONFIG_FILES([libcaes/Makefile])
+dnl AC_CONFIG_FILES([libcstring/Makefile])
+dnl AC_CONFIG_FILES([libcerror/Makefile])
+dnl AC_CONFIG_FILES([libcthreads/Makefile])
+dnl AC_CONFIG_FILES([libcdata/Makefile])
+dnl AC_CONFIG_FILES([libclocale/Makefile])
+dnl AC_CONFIG_FILES([libcdatetime/Makefile])
+dnl AC_CONFIG_FILES([libcnotify/Makefile])
+dnl AC_CONFIG_FILES([libcsplit/Makefile])
+dnl AC_CONFIG_FILES([libuna/Makefile])
+dnl AC_CONFIG_FILES([libcfile/Makefile])
+dnl AC_CONFIG_FILES([libcpath/Makefile])
+dnl AC_CONFIG_FILES([libbfio/Makefile])
+dnl AC_CONFIG_FILES([libfcache/Makefile])
+dnl AC_CONFIG_FILES([libfdata/Makefile])
+dnl AC_CONFIG_FILES([libfvalue/Makefile])
+dnl AC_CONFIG_FILES([libhmac/Makefile])
+dnl AC_CONFIG_FILES([libcaes/Makefile])
 AC_CONFIG_FILES([libewf/Makefile])
-AC_CONFIG_FILES([libodraw/Makefile])
-AC_CONFIG_FILES([libsmdev/Makefile])
-AC_CONFIG_FILES([libsmraw/Makefile])
-AC_CONFIG_FILES([libcsystem/Makefile])
+dnl AC_CONFIG_FILES([libodraw/Makefile])
+dnl AC_CONFIG_FILES([libsmdev/Makefile])
+dnl AC_CONFIG_FILES([libsmraw/Makefile])
+dnl AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([ewftools/Makefile])
 AC_CONFIG_FILES([ewf.net/Makefile])
 AC_CONFIG_FILES([pyewf/Makefile])
diff -ru libewf-5e47a75e30a4dd3068f8210f0a62c37b58329d7f.old/Makefile.am libewf-5e47a75e30a4dd3068f8210f0a62c37b58329d7f.new/Makefile.am
--- libewf-5e47a75e30a4dd3068f8210f0a62c37b58329d7f.old/Makefile.am	2015-01-26 08:31:34.000000000 +0100
+++ libewf-5e47a75e30a4dd3068f8210f0a62c37b58329d7f.new/Makefile.am	2015-06-30 19:41:53.524496055 +0200
@@ -3,28 +3,7 @@
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcdatetime \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfvalue \
-	libhmac \
-	libcaes \
 	libewf \
-	libodraw \
-	libsmdev \
-	libsmraw \
-	libcsystem \
 	ewftools \
 	ewf.net \
 	pyewf \
@@ -79,22 +58,6 @@
 
 library:
 	(cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcstring && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libclocale && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcsplit && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcfile && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcpath && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libbfio && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfcache && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfvalue && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libhmac && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcaes && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libewf && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS))
 
@@ -113,28 +76,7 @@
 	done && test -z "$$fail"
 
 splint:
-	(cd $(srcdir)/libcstring && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdata && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdatetime && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libclocale && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcsplit && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcfile && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcpath && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libbfio && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfcache && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdata && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfvalue && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libhmac && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcaes && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libewf && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libodraw && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libsmdev && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libsmraw && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcsystem && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/ewftools && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/pyewf && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS))
