From 19435220bfa377a503a32aa4b0bb660cfd8a274a Mon Sep 17 00:00:00 2001
From: pancake <pancake@nopcode.org>
Date: Fri, 19 Mar 2021 10:50:52 +0100
Subject: [PATCH] Fix #9 - move unused variable into commented code block

---
 src/core/Cutter.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/Cutter.cpp b/src/core/Cutter.cpp
index af4a2c33..b8f082c4 100644
--- a/src/core/Cutter.cpp
+++ b/src/core/Cutter.cpp
@@ -612,8 +612,8 @@ bool CutterCore::loadFile(QString path, ut64 baddr, ut64 mapaddr, int perms, int
         // Not loading RBin info coz va = false
     }
 
-    auto iod = core->io ? core->io->desc : NULL;
 /*
+    auto iod = core->io ? core->io->desc : NULL;
     auto debug = core->file && iod && (core->file->fd == iod->fd) && iod->plugin && \
                  iod->plugin->isdbg;
 */
