diff -ru libcdata-ed66801e97c516c405486508d280fda3e8114e9d.old/configure.ac libcdata-ed66801e97c516c405486508d280fda3e8114e9d.new/configure.ac
--- libcdata-ed66801e97c516c405486508d280fda3e8114e9d.old/configure.ac	2024-04-14 05:59:58.000000000 +0200
+++ libcdata-ed66801e97c516c405486508d280fda3e8114e9d.new/configure.ac	2024-05-13 15:22:21.960071737 +0200
@@ -94,8 +94,6 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
 AC_CONFIG_FILES([libcdata/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libcdata-ed66801e97c516c405486508d280fda3e8114e9d.old/Makefile.am libcdata-ed66801e97c516c405486508d280fda3e8114e9d.new/Makefile.am
--- libcdata-ed66801e97c516c405486508d280fda3e8114e9d.old/Makefile.am	2024-04-14 05:59:58.000000000 +0200
+++ libcdata-ed66801e97c516c405486508d280fda3e8114e9d.new/Makefile.am	2024-05-13 15:22:21.961071741 +0200
@@ -3,8 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
 	libcdata \
 	po \
 	manuals \
@@ -61,8 +59,6 @@
 
 library:
 	cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS)
-	cd $(srcdir)/libcthreads && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/libcdata && $(MAKE) $(AM_MAKEFLAGS)
 	cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS)
 
