diff -ru libbde-7985ac45c98ae264c4bc2f5854e561db8fcdbda1.orig/configure.ac libbde-7985ac45c98ae264c4bc2f5854e561db8fcdbda1.new/configure.ac
--- libbde-7985ac45c98ae264c4bc2f5854e561db8fcdbda1.orig/configure.ac	2016-04-18 18:04:37.000000000 +0200
+++ libbde-7985ac45c98ae264c4bc2f5854e561db8fcdbda1.new/configure.ac	2016-06-20 20:05:49.518864008 +0200
@@ -216,26 +216,26 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcstring/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libcsplit/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
-AC_CONFIG_FILES([libcpath/Makefile])
-AC_CONFIG_FILES([libbfio/Makefile])
-AC_CONFIG_FILES([libfcache/Makefile])
-AC_CONFIG_FILES([libfdata/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfvalue/Makefile])
-AC_CONFIG_FILES([libhmac/Makefile])
-AC_CONFIG_FILES([libcaes/Makefile])
+dnl AC_CONFIG_FILES([libcstring/Makefile])
+dnl AC_CONFIG_FILES([libcerror/Makefile])
+dnl AC_CONFIG_FILES([libcthreads/Makefile])
+dnl AC_CONFIG_FILES([libcdata/Makefile])
+dnl AC_CONFIG_FILES([libclocale/Makefile])
+dnl AC_CONFIG_FILES([libcnotify/Makefile])
+dnl AC_CONFIG_FILES([libcsplit/Makefile])
+dnl AC_CONFIG_FILES([libuna/Makefile])
+dnl AC_CONFIG_FILES([libcfile/Makefile])
+dnl AC_CONFIG_FILES([libcpath/Makefile])
+dnl AC_CONFIG_FILES([libbfio/Makefile])
+dnl AC_CONFIG_FILES([libfcache/Makefile])
+dnl AC_CONFIG_FILES([libfdata/Makefile])
+dnl AC_CONFIG_FILES([libfdatetime/Makefile])
+dnl AC_CONFIG_FILES([libfguid/Makefile])
+dnl AC_CONFIG_FILES([libfvalue/Makefile])
+dnl AC_CONFIG_FILES([libhmac/Makefile])
+dnl AC_CONFIG_FILES([libcaes/Makefile])
 AC_CONFIG_FILES([libbde/Makefile])
-AC_CONFIG_FILES([libcsystem/Makefile])
+dnl AC_CONFIG_FILES([libcsystem/Makefile])
 AC_CONFIG_FILES([bdetools/Makefile])
 AC_CONFIG_FILES([pybde/Makefile])
 AC_CONFIG_FILES([pybde-python2/Makefile])
diff -ru libbde-7985ac45c98ae264c4bc2f5854e561db8fcdbda1.orig/Makefile.am libbde-7985ac45c98ae264c4bc2f5854e561db8fcdbda1.new/Makefile.am
--- libbde-7985ac45c98ae264c4bc2f5854e561db8fcdbda1.orig/Makefile.am	2016-04-18 18:04:37.000000000 +0200
+++ libbde-7985ac45c98ae264c4bc2f5854e561db8fcdbda1.new/Makefile.am	2016-06-20 20:05:18.893312881 +0200
@@ -3,26 +3,7 @@
 SUBDIRS = \
 	include \
 	common \
-	libcstring \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libcsplit \
-	libuna \
-	libcfile \
-	libcpath \
-	libbfio \
-	libfcache \
-	libfdata \
-	libfdatetime \
-	libfguid \
-	libfvalue \
-	libhmac \
-	libcaes \
 	libbde \
-	libcsystem \
 	bdetools \
 	pybde \
 	pybde-python2 \
@@ -83,24 +64,6 @@
 
 library:
 	(cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcstring && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libclocale && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcsplit && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcfile && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcpath && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libbfio && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfcache && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdatetime && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfguid && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfvalue && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libhmac && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcaes && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libbde && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS))
 
@@ -117,26 +80,7 @@
 	done && test -z "$$fail"
 
 splint:
-	(cd $(srcdir)/libcstring && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdata && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libclocale && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcsplit && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcfile && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcpath && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libbfio && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfcache && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdata && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdatetime && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfguid && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfvalue && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libhmac && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcaes && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libbde && $(MAKE) splint $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcsystem && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/bdetools && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/pybde && $(MAKE) splint $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) splint $(AM_MAKEFLAGS))
