diff -ru libemu-ab48695b7113db692982a1839e3d6eb9e73e90a9/configure.ac libemu-ab48695b7113db692982a1839e3d6eb9e73e90a9.new/configure.ac
--- libemu-ab48695b7113db692982a1839e3d6eb9e73e90a9/configure.ac	2018-03-04 23:11:36.750143559 +0100
+++ libemu-ab48695b7113db692982a1839e3d6eb9e73e90a9.new/configure.ac	2018-03-04 23:06:29.411571103 +0100
@@ -5,10 +5,11 @@
 
 AC_PREREQ(2.59)
 AC_INIT([libemu], [0.2.0], [nepenthesdev@gmail.com])
-AM_INIT_AUTOMAKE([libemu], [0.2.0])
+AM_INIT_AUTOMAKE([subdir-objects])
 AC_REVISION([$Id$])
 
 AC_PREFIX_DEFAULT(/opt/libemu)
+AC_CONFIG_MACRO_DIRS([m4])
 AC_CONFIG_SRCDIR([include/emu/emu.h])
 AM_CONFIG_HEADER([config.h])
 # AM_MAINTAINER_MODE
diff -ru libemu-ab48695b7113db692982a1839e3d6eb9e73e90a9/Makefile.am libemu-ab48695b7113db692982a1839e3d6eb9e73e90a9.new/Makefile.am
--- libemu-ab48695b7113db692982a1839e3d6eb9e73e90a9/Makefile.am	2013-04-10 21:32:25.000000000 +0200
+++ libemu-ab48695b7113db692982a1839e3d6eb9e73e90a9.new/Makefile.am	2018-03-04 23:05:17.886596611 +0100
@@ -1,5 +1,7 @@
 # $Id$
 
+ACLOCAL_AMFLAGS= -I m4
+
 AUTOMAKE_OPTIONS = foreign
 
 SUBDIRS = src include testsuite doc bindings tools
