diff -ru libsmdev-9b9c94244a9d0df752fbb0cbee20207a2ee0c7e1.old/configure.ac libsmdev-9b9c94244a9d0df752fbb0cbee20207a2ee0c7e1.new/configure.ac
--- libsmdev-9b9c94244a9d0df752fbb0cbee20207a2ee0c7e1.old/configure.ac	2024-05-05 10:52:36.000000000 +0200
+++ libsmdev-9b9c94244a9d0df752fbb0cbee20207a2ee0c7e1.new/configure.ac	2024-06-03 01:12:18.501594098 +0200
@@ -140,13 +140,6 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libclocale/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libcfile/Makefile])
 AC_CONFIG_FILES([libsmdev/Makefile])
 AC_CONFIG_FILES([pysmdev/Makefile])
 AC_CONFIG_FILES([smdevtools/Makefile])
diff -ru libsmdev-9b9c94244a9d0df752fbb0cbee20207a2ee0c7e1.old/Makefile.am libsmdev-9b9c94244a9d0df752fbb0cbee20207a2ee0c7e1.new/Makefile.am
--- libsmdev-9b9c94244a9d0df752fbb0cbee20207a2ee0c7e1.old/Makefile.am	2024-05-05 10:52:36.000000000 +0200
+++ libsmdev-9b9c94244a9d0df752fbb0cbee20207a2ee0c7e1.new/Makefile.am	2024-06-03 01:11:45.630509689 +0200
@@ -3,13 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libclocale \
-	libcnotify \
-	libuna \
-	libcfile \
 	libsmdev \
 	smdevtools \
 	pysmdev \
@@ -77,13 +70,6 @@
 
 library:
 	(cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libclocale && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcfile && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libsmdev && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS))
 
