diff -ru libfmapi-b25ce6fd691e744dd090cd2ec754766b42a620e5.orig/configure.ac libfmapi-b25ce6fd691e744dd090cd2ec754766b42a620e5.new/configure.ac
--- libfmapi-b25ce6fd691e744dd090cd2ec754766b42a620e5.orig/configure.ac	2015-01-04 13:31:58.000000000 +0100
+++ libfmapi-b25ce6fd691e744dd090cd2ec754766b42a620e5.new/configure.ac	2015-07-01 02:38:38.988452540 +0200
@@ -142,14 +142,6 @@
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_FILES([include/Makefile])
 AC_CONFIG_FILES([common/Makefile])
-AC_CONFIG_FILES([libcerror/Makefile])
-AC_CONFIG_FILES([libcthreads/Makefile])
-AC_CONFIG_FILES([libcdata/Makefile])
-AC_CONFIG_FILES([libcnotify/Makefile])
-AC_CONFIG_FILES([libuna/Makefile])
-AC_CONFIG_FILES([libfdatetime/Makefile])
-AC_CONFIG_FILES([libfguid/Makefile])
-AC_CONFIG_FILES([libfwnt/Makefile])
 AC_CONFIG_FILES([libfmapi/Makefile])
 AC_CONFIG_FILES([po/Makefile.in])
 AC_CONFIG_FILES([po/Makevars])
diff -ru libfmapi-b25ce6fd691e744dd090cd2ec754766b42a620e5.orig/Makefile.am libfmapi-b25ce6fd691e744dd090cd2ec754766b42a620e5.new/Makefile.am
--- libfmapi-b25ce6fd691e744dd090cd2ec754766b42a620e5.orig/Makefile.am	2015-01-04 13:31:58.000000000 +0100
+++ libfmapi-b25ce6fd691e744dd090cd2ec754766b42a620e5.new/Makefile.am	2015-07-01 02:38:15.175816488 +0200
@@ -3,14 +3,6 @@
 SUBDIRS = \
 	include \
 	common \
-	libcerror \
-	libcthreads \
-	libcdata \
-	libcnotify \
-	libuna \
-	libfdatetime \
-	libfguid \
-	libfwnt \
 	libfmapi \
 	po \
 	manuals \
@@ -62,14 +53,6 @@
 
 library:
 	(cd $(srcdir)/common && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcerror && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcthreads && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcdata && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libcnotify && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libuna && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfdatetime && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfguid && $(MAKE) $(AM_MAKEFLAGS))
-	(cd $(srcdir)/libfwnt && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/libfmapi && $(MAKE) $(AM_MAKEFLAGS))
 	(cd $(srcdir)/po && $(MAKE) $(AM_MAKEFLAGS))
 
