diff -up old/setup.py new/setup.py
--- old/setup.py	2021-02-08 22:14:54.000000000 +0100
+++ new/setup.py	2021-02-09 15:55:53.785892041 +0100
@@ -367,12 +367,14 @@ ext_modules = [
         sources=["src/CAST.c"],
         py_limited_api=True),
     Extension("Crypto.Cipher._raw_des",
-        include_dirs=['src/', 'src/libtom/'],
+        include_dirs=['src/'],
         sources=["src/DES.c"],
+        extra_link_args=["-ltomcrypt"],
         py_limited_api=True),
     Extension("Crypto.Cipher._raw_des3",
-        include_dirs=['src/', 'src/libtom/'],
+        include_dirs=['src/'],
         sources=["src/DES3.c"],
+        extra_link_args=["-ltomcrypt"],
         py_limited_api=True),
     Extension("Crypto.Util._cpuid_c",
         include_dirs=['src/'],
@@ -410,8 +412,9 @@ ext_modules = [
         sources=["src/ARC4.c"],
         py_limited_api=True),
     Extension("Crypto.Cipher._Salsa20",
-        include_dirs=['src/', 'src/libtom/'],
+        include_dirs=['src/'],
         sources=["src/Salsa20.c"],
+        extra_link_args=["-ltomcrypt"],
         py_limited_api=True),
     Extension("Crypto.Cipher._chacha20",
         include_dirs=['src/'],
diff -up ./src/DES.c ./src/DES.c
--- old/src/DES.c	2021-02-08 22:14:54.000000000 +0100
+++ new/src/DES.c	2021-02-09 15:55:53.785892041 +0100
@@ -39,7 +39,7 @@ FAKE_INIT(raw_des3)
 
 /* Include the actial DES implementation */
 #define LTC_NO_PROTOTYPES
-#include "libtom/tomcrypt_des.c"
+#include <tomcrypt.h>
 
 struct block_state {
     symmetric_key sk;
