Description: add GCC hardening.
Author: Joao Eriberto Mota Filho <eriberto@debian.org>
Last-Update: 2015-02-23
Index: aesfix-1.0.1/Makefile
===================================================================
--- aesfix-1.0.1.orig/Makefile
+++ aesfix-1.0.1/Makefile
@@ -1,10 +1,10 @@
-CXXFLAGS= -Wall -O4 -funroll-loops
+CXXFLAGS += -Wall -O4 -funroll-loops
 OBJS= aesfix.o errvect.o
 
 all: aesfix
 
 aesfix: $(OBJS)
-	$(CXX) -o aesfix $(OBJS)
+	$(CXX) $(LDFLAGS) -o aesfix $(OBJS)
 
 clean:
 	@rm -f aesfix *~ \#* $(OBJS)
