From 256445958dc5399f183e3330e3fc9fab16ac86e8 Mon Sep 17 00:00:00 2001
From: Michal Ambroz <723625+xambroz@users.noreply.github.com>
Date: Thu, 23 Nov 2023 08:57:33 +0100
Subject: [PATCH 1/2] else/elif syntax error

---
 officeparser.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/officeparser.py b/officeparser.py
index ca6bbd5..b70a475 100755
--- a/officeparser.py
+++ b/officeparser.py
@@ -879,7 +879,7 @@ def check_value(name, expected, value):
             PROJECTCOMPATVERSION_Size = unpack("<L", dir_stream.read(4))[0]
             check_value('PROJECTCOMPATVERSION_Size', 0x0004, PROJECTCOMPATVERSION_Size)
             PROJECTCOMPATVERSION_CompatVersion = unpack("<L", dir_stream.read(4))[0]
-        else record_check == 0x0002:
+        elif record_check == 0x0002:
             # PROJECTLCID Record
             PROJECTLCID_Id = record_check
             check_value('PROJECTLCID_Id', 0x0002, PROJECTLCID_Id)

From 179e4e636350f5fd3cbcc91183ada3acdc658852 Mon Sep 17 00:00:00 2001
From: Michal Ambroz <723625+xambroz@users.noreply.github.com>
Date: Thu, 23 Nov 2023 09:12:27 +0100
Subject: [PATCH 2/2] if syntax error

---
 officeparser.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/officeparser.py b/officeparser.py
index b70a475..afe7ce9 100755
--- a/officeparser.py
+++ b/officeparser.py
@@ -1163,7 +1163,7 @@ def check_value(name, expected, value):
 
             code_stream = ofdoc.find_stream_by_name(MODULESTREAMNAME_StreamName)
             # check for None
-            if code_stream = None:
+            if code_stream == None:
                 logging.warning('Stream name \"{0}\" not found'.format(MODULESTREAMNAME_StreamName))
                 continue
 
