diff -ru libexe-cab97184be54abde56fbaaddbe2c9147c9f9510c/configure.ac libexe-cab97184be54abde56fbaaddbe2c9147c9f9510c.new/configure.ac
--- libexe-cab97184be54abde56fbaaddbe2c9147c9f9510c/configure.ac	2024-04-20 19:18:22.000000000 +0200
+++ libexe-cab97184be54abde56fbaaddbe2c9147c9f9510c.new/configure.ac	2024-06-03 01:58:27.029703416 +0200
@@ -1,4 +1,4 @@
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 AC_INIT(
  [libexe],
@@ -34,7 +34,7 @@
 
 dnl Support of internationalization (i18n)
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.21])
+AM_GNU_GETTEXT_VERSION([0.19])
 
 dnl Check for compiler language support
 AC_C_CONST
