diff -ru sqlmap/lib/core/settings.py sqlmap.new/lib/core/settings.py
--- sqlmap/lib/core/settings.py	2011-04-11 02:23:42.000000000 +0200
+++ sqlmap.new/lib/core/settings.py	2012-02-05 00:09:29.383630158 +0100
@@ -18,7 +18,8 @@
 
 # sqlmap version and site
 VERSION            = "0.9"
-REVISION           = getRevisionNumber()
+#REVISION           = getRevisionNumber()
+REVISION           = "@@REVISION@@"
 VERSION_STRING     = "sqlmap/%s" % VERSION
 DESCRIPTION        = "automatic SQL injection and database takeover tool"
 SITE               = "http://sqlmap.sourceforge.net"
