diff -ru libfguid-c79feddbafec7d24df8c661e8c9be1417d4dedf4.old/configure.ac libfguid-c79feddbafec7d24df8c661e8c9be1417d4dedf4.new/configure.ac
--- libfguid-c79feddbafec7d24df8c661e8c9be1417d4dedf4.old/configure.ac	2024-04-15 07:25:11.000000000 +0200
+++ libfguid-c79feddbafec7d24df8c661e8c9be1417d4dedf4.new/configure.ac	2024-05-19 01:00:57.257011114 +0200
@@ -1,4 +1,4 @@
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 
 AC_INIT(
  [libfguid],
@@ -33,7 +33,7 @@
 
 dnl Support of internationalization (i18n)
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.21])
+AM_GNU_GETTEXT_VERSION([0.19])
 
 dnl Check for compiler language support
 AC_C_CONST
