diff -ru impacket-0.11.0.orig/impacket/examples/mssqlshell.py impacket-0.11.0.new/impacket/examples/mssqlshell.py
--- impacket-0.11.0.orig/impacket/examples/mssqlshell.py	2023-07-31 18:06:55.000000000 +0200
+++ impacket-0.11.0.new/impacket/examples/mssqlshell.py	2024-01-24 01:49:04.682476795 +0100
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 # Impacket - Collection of Python classes for working with network protocols.
 #
 # Copyright (C) 2023 Fortra. All rights reserved.
